metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -306,8 +306,11 @@ ReplicaPlacer replicaPlacer() {
      */
     public void activate() {
         heartbeatManager = new BrokerHeartbeatManager(logContext, time, sessionTimeoutNs);
+        long nowNs = time.nanoseconds();
         for (BrokerRegistration registration : brokerRegistrations.values()) {
             heartbeatManager.register(registration.id(), registration.fenced());
+            heartbeatManager.tracker().updateContactTime(
+                new BrokerIdAndEpoch(registration.id(), registration.epoch()), nowNs);
```

Comment:
nit: may want to consider only adding unfenced brokers?

adding fenced brokers to the tracker may delay fencing a newly failed broker by whatever is the session timeout.

Reply:
I think it's OK to add them here just for consistency with the `processBrokerHeartbeat` code, which also adds them. It shouldn't delay anything; the tracker just manages last contact times.

In the case where both a fenced and an unfenced broker had timed out, we would remove the fenced one from the tracker, do nothing, and then reschedule `maybeFenceOneStaleBroker` immediately as the next controllerr event. So there is no delay really.

Reply:
sounds good.